### PR TITLE
typeInformation contain global config values

### PR DESCRIPTION
**Pull request by KeshavSoni2511**
Fix for issue #1515

### CODE REVIEW

**Comment by KeshavSoni2511** [comment]
> Could you please include the simplification of blocks referenced in [#1515 (comment)](https://github.com/telefonicaid/iotagent-node-lib/issues/1515#issuecomment-1923480572) ?

Hi @AlvaroVega, I have done & tested the simplication in https://github.com/telefonicaid/iotagent-node-lib/blob/d1586fb65a4848ba615ad0cf0da7e39afb2fa350/lib/services/ngsi/entities-NGSI-v2.js#L304-L309

to

Screenshot 2024-02-16 174748


but test case [httpBindings-test.js](https://github.com/telefonicaid/iotagent-ul/blob/3dd03a5c62ac16b38ccf964e49a9c0fba8ce765c/test/unit/ngsiv2/httpBindings-test.js#L621) was failing.

The issue can be fixed by adding attribute ```"timestamp": "true" ``` in [provisionDeviceProduction.json](https://github.com/telefonicaid/iotagent-ul/blob/master/test/deviceProvisioning/provisionDeviceProduction.json). In case of **iotagent-json** it works without any test case failure.

Please confirm if this change is required or you have any other suggestion.

**Comment by KeshavSoni2511** [comment]
> > Could you please include the simplification of blocks referenced in [#1515 (comment)](https://github.com/telefonicaid/iotagent-node-lib/issues/1515#issuecomment-1923480572) ?
> 
> Hi @AlvaroVega, I have done & tested the simplication in
> 
> https://github.com/telefonicaid/iotagent-node-lib/blob/d1586fb65a4848ba615ad0cf0da7e39afb2fa350/lib/services/ngsi/entities-NGSI-v2.js#L304-L309
> 
> to
> 
> Screenshot 2024-02-16 174748
> 
> but test case [httpBindings-test.js](https://github.com/telefonicaid/iotagent-ul/blob/3dd03a5c62ac16b38ccf964e49a9c0fba8ce765c/test/unit/ngsiv2/httpBindings-test.js#L621) was failing.
> 
> The issue can be fixed by adding attribute `"timestamp": "true" ` in [provisionDeviceProduction.json](https://github.com/telefonicaid/iotagent-ul/blob/master/test/deviceProvisioning/provisionDeviceProduction.json). In case of **iotagent-json** it works without any test case failure.
> 
> Please confirm if this change is required or you have any other suggestion.


Hi @AlvaroVega, please review my understanding. Thanks

**Comment by fgalan** [comment]
> Hi @AlvaroVega, please review my understanding. Thanks

It seems there is a conflict on CHANGES_NEXT_RELEASE. This has to be solved (it's easy :) before merging.

**Comment by KeshavSoni2511** [comment]
> Good news about iotagent-json tests works without any modifications after simplifying checks in iotagent-node-lib. I've see not problem in perform another PR to modify iotagent-ul tests in the way you describe (change timestamp to false in https://github.com/telefonicaid/iotagent-ul/blob/master/test/deviceProvisioning/provisionDeviceProduction.json and complete current PR with the simplifications in related checks

Hi @AlvaroVega, I have performed **iotagent-ul** test with ```"timestamp": "true" ``` in https://github.com/telefonicaid/iotagent-ul/blob/master/test/deviceProvisioning/provisionDeviceProduction.json as global config value contain ```"timestamp": "true" ```  and test case passed with the proposed change. Please confirm my understanding if I can proceed further and raise another PR on iotagent-ul. Please confirm if you have any other suggestion. Thanks

**Comment by AlvaroVega** [comment]
> > Good news about iotagent-json tests works without any modifications after simplifying checks in iotagent-node-lib. I've see not problem in perform another PR to modify iotagent-ul tests in the way you describe (change timestamp to false in https://github.com/telefonicaid/iotagent-ul/blob/master/test/deviceProvisioning/provisionDeviceProduction.json and complete current PR with the simplifications in related checks
> 
> Hi @AlvaroVega, I have performed **iotagent-ul** test with `"timestamp": "true" ` in https://github.com/telefonicaid/iotagent-ul/blob/master/test/deviceProvisioning/provisionDeviceProduction.json as global config value contain `"timestamp": "true" ` and test case passed with the proposed change. Please confirm my understanding if I can proceed further and raise another PR on iotagent-ul. Please confirm if you have any other suggestion. Thanks

Yes, you can proced it with another PR for iotagent-ul

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1,2 +1,3 @@
+- Fix: typeInformation contain global config values (#1515)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:1]: Taking into account this is an internal refactor to simplify code more than an actual fix, I'd suggest to use this wording better:

```suggestion
- Hardening: simplify implementation so typeInformation contain global config values (#1515)
```

**Review by fgalan — APPROVED** [review]
LGTM

**Comment by fgalan** [comment]
Thanks for the contribution!

**Pull request merged**